### DISCUSSION TITLE
exception raised in invalid_mods_crash_detour. can anyone help me know what it is?

### DISCUSSION THREAD

**Discussion by lonelybud** [YimMenu/YimMenu#2390] [General]
i was playing gta with my own fork of yim menu injected. I also attached vs code debugger to GTA5.exe.

When i was joining a session, i saw these exception and game crashed. Just wanted to know some useful information from developers. 

![Screenshot 2023-11-06 211408](https://github.com/YimMenu/YimMenu/assets/139460769/3fcb6256-b7c8-49e6-98c5-a6bfdeda0e10)
![Screenshot 2023-11-06 211352](https://github.com/YimMenu/YimMenu/assets/139460769/befd2dfd-001d-41d9-92a3-594dd9830269)

while i played the debugger exceptions were seen again at same line but different addresses so i have to disconnect the debugger and let the game crash.

Edit: 
there were no exceptions or warnings present in cout.log.

**Comment by Yimura** [comment]
Yeah this hook is kind off hard to debug on why it's crashing, I've heard it could literally crash for doing something wrong that doesn't seem relevant at all.

You'll have to tell us more about what sort of features you've implemented and if those might be causing the issues.
Or give us more off a stack trace than just this... (including the registers).
Idk, it's sort of like your problem as this is a modified version of YimMenu, if you'd like help you'll have to post some more information.

**Reply by lonelybud** [reply]
Btw i will check for faulty features. How can I get the stack trace?

**Reply by Yimura** [reply]
Well you’re in the debugger view already so on the left side in Visual Studio code there should be something to show the stacktrace. That shows the code flow until the crash occurred.

**Reply by lonelybud** [reply]
thanks

**Reply by lonelybud** [reply]
![image](https://github.com/YimMenu/YimMenu/assets/139460769/0a4a428b-5c2f-440f-8654-97a39cde30fb)

not much info i can see

**Reply by lonelybud** [reply]
GTA5.exe!00007ff6d1d0b621() (Unknown Source:0)
GTA5.exe!00007ff6d1d3875f() (Unknown Source:0)
GTA5.exe!00007ff6d125e4e7() (Unknown Source:0)
YimMenu.dll!big::hooks::invalid_mods_crash_detour(__int64 a1, __int64 a2, int a3, char a4) Line 10 (d:\work\dynamoMenu\src\hooks\protections\invalid_mods_crash_detour.cpp:10)
GTA5.exe!00007ff6d0eaf4bf() (Unknown Source:0)
GTA5.exe!00007ff6d1fd9fc5() (Unknown Source:0)
GTA5.exe!00007ff6d1fd9ee2() (Unknown Source:0)
GTA5.exe!00007ff6d1fb4779() (Unknown Source:0)
GTA5.exe!00007ff6d1fb5116() (Unknown Source:0)
GTA5.exe!00007ff6d1fbe00a() (Unknown Source:0)
GTA5.exe!00007ff6d1c6e732() (Unknown Source:0)
kernel32.dll!00007ffb3f88257d() (Unknown Source:0)
ntdll.dll!00007ffb41aeaa78() (Unknown Source:0)

**Reply by lonelybud** [reply]
[11-06-2023-22-09-25_cout.log](https://github.com/YimMenu/YimMenu/files/13270270/11-06-2023-22-09-25_cout.log)

**Reply by lonelybud** [reply]
i am not quite sure what the reason could be. May be this code was faulty -

1. `self::last_veh = PLAYER::GET_PLAYERS_LAST_VEHICLE();` in `void looped::system_self_globals()`
2. `if (VEHICLE::IS_TOGGLE_MOD_ON(self::last_veh, 18))`
3. ...

problem seems to go away now in new release.

**Comment by lonelybud** [comment]
can debug builds be detected by game or can it crash as opposed to release builds. Its not related to this topic but just asking
